tests/test_http_parser.py

```diff
@@ -921,13 +921,14 @@ async def test_parse_eof_payload(self, stream: Any) -> None:
         p.feed_eof()
 
         assert out.is_eof()
-        assert [(bytearray(b"data"), 4)] == list(out._buffer)
+        assert [(b"data",)] == [(data,) for (data, _) in out._buffer]
 
     async def test_parse_no_body(self, stream: Any) -> None:
         out = aiohttp.FlowControlDataQueue(
             stream, 2**16, loop=asyncio.get_event_loop()
         )
         p = HttpPayloadParser(out, method="PUT")
+        p.feed_eof()
```

Comment:
I'm unclear why these 2 need to change?

Reply:
This tests  have failed, and I thought, since we removed the `size` arg, then we need to little changed the assert in the test 🤔

Reply:
Maybe, but I'd like to understand why these were not needed before. It's not obvious how these changes relate to a size parameter being removed, and could be an indication that some behaviour may break for users.

Reply:
Maybe, but what? We need to understand exactly why the change is needed.

Reply:
I've looked into this, and that doesn't seem to be correct.
The buffer these tests refer to are `streams.DataQueue`, where the size parameter is still expected:
https://github.com/DavidRomanovizc/aiohttp/blob/master/aiohttp/streams.py#L559

So, maybe we want to drop the size parameter from that class as well? Then these changes can be reverted (only dropping the `4` from the test comparison).